Blending jargon

PatrykSzwer opened the pull request:
For https://github.com/Starcounter/RebelsLounge/issues/227
Those are my initial changes to docs according to [Blending jargon document](https://docs.google.com/spreadsheets/d/1RvdpVmlsuDFyUfXbIZnAdD6xTmkmjRNLprPMyorQBGU/edit#gid=0)

My comments about those changes:
- I didn't change current "partial" occurrences, because for me, in most places "partial" is the only word that actually fits there, and I can't see the point in the document that fits these. (most occurrences of the partial word is in the html-views.html document, e.g.: `A partial HTML view is a view that has been nested as a part of a bigger view.`)
- I'm not sure how should I change occurrences of blending point words in this sentence: (from html-views.html)
`2. Through a blending point. A blending point is a combination of a `Self.GET` call, which acts as the partial view-model blending point, and a `<starcounter-include>` tag, which acts as a partial HTML view blending point. The `<starcounter-include>` tag determines where in the DOM tree the partial HTML view should be rendered.`

Please give me some tips what should I do about these cases.

PatrykSzwer committed:
Replaced blending rule with attachment rule

PatrykSzwer committed:
Replaced view with view-model

PatrykSzwer committed:
Changed name of navigation link for view attaching to view-model attaching

PatrykSzwer committed:
Replaced word layout on presentation

PatrykSzwer committed:
Replaced blending point with blended view-model insertion

ErlendLandro commented:
> I didn't change current "partial" occurrences, because for me, in most places "partial" is the only word that actually fits there, and I can't see the point in the document that fits these.  

I agree, the use of "partial" in the documentation was fixed some time ago

I don't know what to do about the second case. That would be up to @warpech to determine

ErlendLandro reviewed (approved):
Looks good to me!

warpech commented:
Can you solve the conflict with `master`?

I will fix the description fo the blending point.

warpech commented:
> 2. Through a blending point. A blending point is a combination of a `Self.GET` call, which acts as the partial view-model blending point, and a `<starcounter-include>` tag, which acts as a partial HTML view blending point. The `<starcounter-include>` tag determines where in the DOM tree the partial HTML view should be rendered.

#### Should be

2. Through a pair of insertion points in a view-model and a view:
   - a blended view-model insertion point is represented by a property in the view-model. The property inserts the result of a `Self.GET` call to a partial view-model in a parent view-model.
   - a blended view insertion point is represented by a `<starcounter-include>` tag in HTML. The tag inserts the partial  view in the parent DOM tree.

PatrykSzwer committed:
Merge branch '2.4' of https://github.com/Starcounter/Docs into blending-jargon

# Conflicts:
#	docs/SUMMARY.md

PatrykSzwer commented:
Conflicts are fixed.

>Should be
Through a pair of insertion points in a view-model and a view:
a blended view-model insertion point is represented by a property in the view-model. The property inserts the result of a Self.GET call to a partial view-model in a parent view-model.
a blended view insertion point is represented by a <starcounter-include> tag in HTML. The tag inserts the partial view in the parent DOM tree.

this part is also done.

PatrykSzwer committed:
Changed description of second way of accessing a partial

warpech committed:
put the explanation of client side libraries included in Starcounter in one place

warpech committed:
update the usage of current versions of the libraries

Polymer 1.0 -> 2.0
Palindrom-polymer-client > Palindrom-client
Puppet-client > Palindrom-client

warpech committed:
the property in starcounter-include is now called `view-model`

warpech committed:
In HTML5, you never need "/>"

It is enough to use ">"

warpech committed:
make better use of the new jargon, rewrite few sentences to be easier to comprehend

warpech commented:
I pushed a few improvements but still have to go through some of the most important files:

```
* [View-model Attaching](topic-guides/blendable-web-apps/view-attaching.md)
* [View Composing](topic-guides/blendable-web-apps/view-composing.md)
* [HTML Views Composition Guidelines](topic-guides/blendable-web-apps/html-views-composition-guidelines.md)
* [Avoiding CSS conflicts](topic-guides/blendable-web-apps/avoiding-css-conflicts.md)
* [Debugging](topic-guides/blendable-web-apps/debugging.md)
```

Thanks @PatrykSzwer for triggering this :)

ErlendLandro commented on docs/topic-guides/blendable-web-apps/web-components.md:
````diff
@@ -10,48 +10,43 @@ Web Components are a set of new features in HTML 5 and beyond. Each feature is u
 
 Web Components are loaded by browser with HTML Imports. HTML Import for Polymer in Starcounter should look like this:
 
-```markup
````
Are you sure it should be `html` and not `markup`?

warpech replied:
I am not sure, there is no info about it on https://betadocs.gitbook.com/features/markdown :(

`html` is what I always used in GitHub and Wordpress. But I agree I should keep what was there. I will revert this change.

warpech committed:
revert change of ```html to ```markup

because I changed that without checking if it works

warpech committed:
replace Bootstrap with Uniform.css

because now Uniform.css is used as the CSS framework for consistent look and feel
remove excessive example of an app shell from the Middleware page

warpech committed:
use the term attachment rule more often

warpech committed:
use the word presentation more often

warpech committed:
defined is a better word

because it is included by "starcounter-include", isn't it

warpech commented:
@Mackiovello I did a bunch of more changes regarding the jargon. Could you please give it a final round of review.

ErlendLandro commented on docs/topic-guides/blendable-web-apps/html-views.md:
````diff
@@ -26,91 +26,97 @@ A partial HTML view is a view that has been nested as a part of a bigger view.
 ```
 
-This handler only returns the JSON and HTML if the app uses the `HtmlFromJsonProvider` and `PartialToStandaloneHtmlProvider` [middleware](../network/middleware.md).
+The view-model will be rendered by the Starcounter app shell only if the app uses the `HtmlFromJsonProvider` and `PartialToStandaloneHtmlProvider` [middleware](../network/middleware.md).
````
Passive form.

The view-model will be rendered by -> The Starcounter app shell renders the view-model if...

ErlendLandro commented on docs/topic-guides/blendable-web-apps/client-side-stack.md:
```diff
@@ -20,29 +20,51 @@ The implementation of JSON-Patch is provided by `json-patch-duplex.js`, which co
-
+## Client side libraries distributed with Starcounter
+
+The `StaticFiles` folder from Starcounter installation is automatically served as a static content folder. When Starcounter server receives a request for a static file, it searches for the file in all of the static content folders. The project folder has higher priority over internal folder.
```
The project folder has higher priority over internal folder. -> The project folder has a higher priority than the internal folder.

ErlendLandro commented on docs/topic-guides/blendable-web-apps/sessions.md:
```diff
@@ -2,7 +2,7 @@
 ## Introduction
 
-Sessions are used to retain the state in your app. A session is represented by an instance of the `Session` class. Together with an instance of a `Json` object it can be used to enable client-server communication and synchronization using JSON patch.
+Sessions are used to retain the state in your app. In Starcounter, the state is represented by a view-model, which is technically an instance of the `Json` class. A session is represented by an instance of the `Session` class. The combination of a session and a view-model enables client-server communication using JSON Patch.
```
> which is technically an instance of the `Json` class

"technically" is ambiguous. Please clarify or remove "technically"

ErlendLandro commented on docs/topic-guides/blendable-web-apps/starcounter-mvvm.md:
```diff
@@ -32,11 +32,11 @@ Typed JSON allows dynamic creation of JSON trees. In addition, Typed JSON allows
 ## View
 
-Because view-models are shared between the client and server, view rendering is performed on the client. In this way, you get better performance and as the number of users grow, so does the number of CPUs that renders the user interfaces.
+Because view-models are shared between the client and server, it allows for the offset the view rendering to be performed on the client. In this way, you get better performance and as the number of users grow, so does the number of CPUs that renders the user interfaces.
```
> it allows for the offset the view rendering to be performed on the client.

This sentence is hard to read. Would something like this work? 

> Because view-models are shared between the client and server, the client can render the view.

I'm not sure if that's what you're trying to say though.

warpech replied:
There was a grammatical error. I will write:

> Because view-models are shared between the client and server, it allows offsetting the view rendering to the client.

My idea was to be more clear, not less clear than the original text.

ErlendLandro commented on docs/topic-guides/blendable-web-apps/view-attaching.md:
```diff
@@ -1,42 +1,44 @@
+6. The server sends the response, which now contains view-models from multiple apps, back to the client.
 
-By using tokens, the apps don't need to know anything about other apps in the code host - they don't even need to know if there are other apps -  they only have to communicate what concept the handlers deal with. Thus, apps should be built to not depend on, or expect, attaching. 
+By using attchment rules, the apps don't need to know anything about other apps in the code host - they don't even need to know if there are other apps -  they only have to communicate what concept the handlers deal with. The apps should be built to not depend on, but expect, attaching.
```
> The apps should be built to not depend on, but expect, attaching.

The meaning of this sentence changed by exchanging "or" for "but". Should apps expect attaching?

warpech replied:
I think that yes, whenever they use `Self.GET` they need to expect attaching.

ErlendLandro replied:
Okay, then this is fine

ErlendLandro commented on docs/topic-guides/blendable-web-apps/web-components.md:
```diff
@@ -63,59 +58,29 @@ How does the browser know which of the components has already been imported then
+#### .bowerrc {#.bowerrc}
 
-Keep in mind that you can use another version of Polymer by putting it into your local `sys` folder. This will affect all other apps, though.
+Keep in mind that you can use another version of Polymer by putting it into your local `sys` folder. This will affect all other apps, though. To prevent that, a `.bowerrc` file is used.
```
> To prevent that, a `.bowerrc` file is used.

Passive form

To prevent that, use a `.bowerrc` file

warpech committed:
Merge branch '2.4' into blending-jargon

# Conflicts:
#	docs/topic-guides/blendable-web-apps/html-views-composition-guidelines.md

warpech committed:
apply improvements suggested by @Mackiovello in https://github.com/Starcounter/Docs/pull/203

ErlendLandro reviewed (approved):
LGTM!

warpech commented:
Great, thanks both of you!
